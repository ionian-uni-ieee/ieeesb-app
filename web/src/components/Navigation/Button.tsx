import React from 'react'
import { Link } from 'react-router-dom'
import { scrollToID } from 'utils/browserHelpers'

interface IButtonProps {
  label: string
  href?: string
  onClick?(e: React.MouseEvent<HTMLAnchorElement>): void
}

const Button: React.FC<IButtonProps> = props => {
  const {
    label,
    href,
    onClick= null,
  } = props

  const handleClick = (e: React.MouseEvent<HTMLAnchorElement>): void => {
    if (onClick) {
      onClick(e)
    }
    if (!href) {
      return
    }
    e.preventDefault()
    scrollToID(href)
  }

  return (
    <Link
      to={href || 'ieee'}
      className='ui navigation__button'
      onClick={handleClick}
    >
      {label}
    </Link>
  )
}

export default Button
