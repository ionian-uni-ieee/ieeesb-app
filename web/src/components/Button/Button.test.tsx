import React from 'react'

import { fireEvent, render } from '@testing-library/react'

import Button from './Button'

describe('Button', () => {
  it('should render', () => {
    const { container } = render(<Button />)
    expect(container).toMatchSnapshot()
  })
})
