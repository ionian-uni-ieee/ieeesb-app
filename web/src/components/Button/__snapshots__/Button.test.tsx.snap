// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
<div>
  <div
    class="
        button
        
        
        undefined
      "
  >
    <button
      class="button__element"
    />
  </div>
</div>
`;
